wiki-site/docs/Data Acquisition/Custom DAQ PCB Boards/DAQ Revision 1/Revision 1.md

```diff
@@ -0,0 +1,260 @@
+Note that this external oscillator needs to be within a certain range of frequencies and this differs for each board[^2]. The stm32 F429 that is used for this DAQ can take an oscillator frequency ranging from 4 to 26 MHz.
+
+### VDD, VDDA, VSS, VSSA, and Decoupling Capacitors
+
+- `VDD` — the digital voltage supply to power digital devices, such as GPIOs.
+- `VDDA` — the analog voltage supply to power analog devices, such as ADCs.
+- `VSS` and `VSSA` — the respective grounds.
+
+To ensure that the voltage supplied to the microcontroller has relatively low noise levels, decoupling capacitors are used to filter the voltage supply.
+
+!!! success
+    The rule of thumb here is to have 1 decoupling capacitor for each `VDD`, `Vbat`, or `VDDA` pin that is present on the microcontroller.
+
+`VDDA` and `VDD` tend to be filtered a little differently so they tend to require their specific values for filtering. Typically, a 100 nF decoupling capacitor is used for each pin. A 4.7 uF bulk decoupling capacitor is added to the `VDD` pin and a 1 uF decoupling capacitor is added to the analog pin filtering.
+
+!!! example
+
+    This DAQ contains 13 VDD pins (this includes `Vbat`) and 1 VDDA pin. So you would have the following:
+    - Fourteen 100 nF decoupling capacitors
+    - One 4.7 uF bulk decoupling capacitor for `VDD`
+    - One 1 uF bulk decoupling capacitor for the `VDDA`.
+
+
+
+## Communication Protocols
+
+!!! info
+    The pinouts for supporting specific communication protocols were selected with CubeMX as a visual aid.
+
+### SPI 
+
+Two SPI peripherals are supported. These is rather straightforward as the pins are directly connected to the MCU.
+
+For the first SPI peripheral (that is, `SPI3` on CubeMX), the pins are:
+
+| Microcontroller Pins | SPI Functionality |
+| :------------------: | :---------------: |
+|         PC12         |       `MOSI`      |
+|         PC11         |       `MISO`      |
+|         PC10         |       `SCK`       |
+|         PG9          |       `CS`        |
+
+
+For the second SPI peripheral (that is, `SPI5` on CubeMX), the pins are:
+
+| Microcontroller Pins | SPI Functionality |
+| :------------------: | :---------------: |
+|         PF9          |       `MOSI`      |
+|         PF8          |       `MISO`      |
+|         PF7          |       `SCK`       |
+|         PF6          |       `CS`        |
+
+
+### I2C
+I2C requires a pull-up resistor to function properly. These resistors typically range from 2.2 Kilo-Ohm to 10 Kilo-Ohms. The values you consider for the pull resistors matter because they influence the speed and frequency you can support for I2C communication. 
+
+- If 100 KHz is fast enough for your applications, then you can use a 10 Kilo-Ohm pull-up resistor.
+- If 400 KHz, then 2.2 Kilo-Ohm pull-up resistors will be preferable.
```

Comment:
How did you correlate the resistance values to these speeds?

Reply:
these are rule of thumb values for i2c this link talks about them
https://docs.particle.io/reference/device-os/api/wire-i2c/pull-up-resistors-i2c/